src/path.c

```diff
@@ -1887,7 +1887,7 @@ extern int git_path_is_gitfile(const char *path, size_t pathlen, git_path_gitfil
 	const char *file, *hash;
 	size_t filelen;
 
-	if (gitfile < 0 && gitfile >= ARRAY_SIZE(gitfiles)) {
+	if (!(gitfile >= GIT_PATH_GITFILE_GITIGNORE && gitfile < ARRAY_SIZE(gitfiles))) {
```

Comment:
I normally wouldn't like relying on the fact that this enum is actually the first one. But there already is a comment in "src/path.h", stating that the enum is sensitive to ordering. So this is fine

Reply:
I'm also quite sure there's no simple way to check for an enum signedness, without resorting to build-system shenanigans or explicitly making the enum signed. Hence the test case I've added, which should hopefully catch a change in definition.